apis/duck/v1/source_types_test.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"context"
+	"testing"
+
+	"knative.dev/pkg/apis"
+)
+
+func TestSourceValidate(t *testing.T) {
+	for _, tt := range []struct {
+		name string
+		src  *Source
+		want *apis.FieldError
+	}{{
+		name: "empty source validation",
+		src:  nil,
+		want: nil,
+	}, {
+		name: "empty source ceOverrides extensions validation",
+		src: &Source{Spec: SourceSpec{
+			CloudEventOverrides: &CloudEventOverrides{Extensions: map[string]string{}},
+		}},
+		want: nil,
+	}, {
+		name: "empty extension name error",
+		src: &Source{Spec: SourceSpec{
+			CloudEventOverrides: &CloudEventOverrides{Extensions: map[string]string{"": "test"}},
+		}},
+		want: apis.ErrInvalidKeyName(
+			"",
+			"spec.ceOverrides.extensions",
+			"keys MUST NOT be empty",
+		),
+	}, {
+		name: "extension name too long",
+		src: &Source{Spec: SourceSpec{
+			CloudEventOverrides: &CloudEventOverrides{
+				Extensions: map[string]string{"nameLongerThan20Characters": "test"},
+			},
+		}},
+		want: nil,
```

Comment:
The 20-chars limit is just a recommendation. It is valid to have extensions that are longer than 20 characters.

Reply:
Yup this test is here just to ensure it passes, and maybe in the future to verify we return a warning at least :)

Reply:
Sounds reasonable!
Would you then consider renaming the test to `long extension name is valid` or something like that?

Reply:
Sure 👍🏼